src/server/pkg/metrics/metrics.go

```diff
@@ -1,64 +1,125 @@
 package metrics
 
 import (
+	"fmt"
 	"time"
 
+	"github.com/pachyderm/pachyderm/src/client/pkg/config"
 	"github.com/pachyderm/pachyderm/src/client/pkg/uuid"
 	"github.com/pachyderm/pachyderm/src/client/version"
 	db "github.com/pachyderm/pachyderm/src/server/pfs/db"
 
 	"github.com/dancannon/gorethink"
+	"github.com/segmentio/analytics-go"
 	"go.pedge.io/lion/proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
 	kube "k8s.io/kubernetes/pkg/client/unversioned"
 )
 
-func dbMetrics(dbClient *gorethink.Session, pfsDbName string, ppsDbName string, metrics *Metrics) {
+var globalReporter *Reporter
```

Comment:
Neither of those strike me as good reasons to make `Reporter` global. You can still check if metrics are enabled and re-use the segment client, you just have to explicitly pass the `Reporter` instance to the pieces of code that are using it. That should only be the 2 api_servers both of which are instantiated in pachd's main function so it doesn't seem like it'd be that big of a change.

I'm a bit confused about what you mean by "both of those metrics flags." What are the 2 flags?